pkg/webhook/propagationpolicy/validating.go

```diff
@@ -60,6 +60,17 @@ func (v *ValidatingAdmission) Handle(_ context.Context, req admission.Request) a
 			klog.Error(err)
 			return admission.Denied(err.Error())
 		}
+
+		if policy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel] !=
+			oldPolicy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel] {
+			return admission.Denied(fmt.Sprintf("label %s is immutable, it can only be set by the system during creation",
+				policyv1alpha1.PropagationPolicyPermanentIDLabel))
+		} else if req.Operation == admissionv1.Create {
```

Comment:
Isn't that the wrong level of judgment?

Reply:
Yeah, just check the label would be fine, don't have to care what the operation is. (don't need that `else if`)
```
	if _, exist := policy.Labels[policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel]; !exist {
		return admission.Denied(fmt.Sprintf("label %s is required, it should be set by the mutating admission webhook during creation",
			policyv1alpha1.ClusterPropagationPolicyPermanentIDLabel))
	}
```

Reply:
Just like the following `validation.ValidatePropagationSpec(policy.Spec)`.

Reply:
feel unnecessary to check twice when the operation is `UPDATE`.

Reply:
?
Why do you think it checks twice?
In the condition `req.Operation == admissionv1.Update`, we only foucs on the label can not be updated, consider the situation that mutating admission webhook might not work for whatever reason, both `old` and `new` might not have the label, so it can pass the check.

But we do require the label to be set, so we need to check it, but this time we focus on the existence.

Reply:
if the mutating admission webhook doesn't work, the policy cannot be created cause we check it when creating. 
> both old and new might not have the label, so it can pass the check.

so it will never happen.

when `req.Operation == admissionv1.Update`, it will only pass the check if there is no problem with the policy label. so we don't need check it again.

Reply:
Update like this:
```diff
diff --git a/pkg/webhook/propagationpolicy/validating.go b/pkg/webhook/propagationpolicy/validating.go                      
index 2e7c02c8e..98a73cecd 100644                                                                                           
--- a/pkg/webhook/propagationpolicy/validating.go                                                                           
+++ b/pkg/webhook/propagationpolicy/validating.go                                                                           
@@ -48,6 +48,10 @@ func (v *ValidatingAdmission) Handle(_ context.Context, req admission.Request) a                         
        }                                                                                                                   
        klog.V(2).Infof("Validating PropagationPolicy(%s/%s) for request: %s", policy.Namespace, policy.Name, req.Operation)
                                                                                                                            
+       if _, exist := policy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel]; !exist {
+               return admission.Denied(fmt.Sprintf("label %s is required, it should be set by the mutating admission webhook during creation",
+                       policyv1alpha1.PropagationPolicyPermanentIDLabel))
+       }
        if req.Operation == admissionv1.Update {
                oldPolicy := &policyv1alpha1.PropagationPolicy{}
                err = v.Decoder.DecodeRaw(req.OldObject, oldPolicy)
@@ -65,11 +69,6 @@ func (v *ValidatingAdmission) Handle(_ context.Context, req admission.Request) a
                        oldPolicy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel] {
                        return admission.Denied(fmt.Sprintf("label %s is immutable, it can only be set by the system during creation",
                                policyv1alpha1.PropagationPolicyPermanentIDLabel))
-               } else if req.Operation == admissionv1.Create {
-                       if _, exist := policy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel]; !exist {
-                               return admission.Denied(fmt.Sprintf("label %s is required, it should be set by the mutating admission webhook during creation",
-                                       policyv1alpha1.PropagationPolicyPermanentIDLabel))
-                       }
                }
        }

```

Reply:
I get what you mean.  And you are right, that case i mentioned will not happen.

But, I tend to a simple pattern that lets validating admission webhook not care what exactly the operation is, and just focus on the validation.
For those checks really need to compare the `old` and `new`, we can put all these checks together. 

The pattern is simple and readable.